umbral/curve.py

```diff
@@ -0,0 +1,63 @@
+from umbral import openssl
+
+
+class Curve:
+    """
+    Acts as a container to store constant variables such as the OpenSSL
+    curve_nid, the EC_GROUP struct, and the order of the curve. This also acts
+    as a convenient whitelist to limit the curves used in pyUmbral.
+    """
+
+    __AVAIL_CURVES = {
+        'secp256r1': 415,
+        'secp256k1': 714,
+        'secp384r1': 715,
+    }
+
+    def __init__(self, curve_nid: int):
+        """
+        Instantiates an OpenSSL curve with the provided curve_nid and derives
+        the proper EC_GROUP struct and order. You can _only_ instantiate curves
+        with supported nids (see `Curve.supported_curves`).
+        """
+        if curve_nid not in self.__AVAIL_CURVES.values():
```

Comment:
Looks like this is intentionally used by NID only?

Reply:
Yep, it's the method to creating an `EC_GROUP` struct. All the other calls require the `EC_GROUP` and this is the only place to get it. I'm evaluating whether or not we even need to store the `curve_nid` to prevent misuse.